axon-application-development/event-handling/introduction.md

```diff
@@ -0,0 +1,76 @@
+
+The[ Event Dispatchers ](dispatching-events.md)section details the capabilities that the Axon Framework provides to help in the development of these Components.
+
+#### _Event Handlers_
+
+Event Handler Components are the components that act on published events i.e. they define the business logic to be performed when an Event is received
+
+Usually, this involves,
+
+* Updating View Models
+* Trigger External Activities \(e.g. Third Party Integrations\) 
+* Manage Complex Transactions \(Sagas\)
+
+The [Event Handlers](handling-events.md) section details the capabilities that the Axon Framework provides to help in the development of these Components.
+
+A summary of the representation of the functional components is depicted below.
+
+![](../../.gitbook/assets/axon-event-components.jpg)
+
+### Non-Functional Components
+
+The Non-Functional Components help address the _**technical concerns**_ around Event Publishing and Handling i.e. the "_**How**_" \(e.g. Which is the protocol we need to use to publish/deliver Event Messages ? How should errors be handled ? How do we implement the threading mode for processing Events ? How should we sequence the processing of Events?\) 
+
+The Axon Framework provides the following  Non-Functional components.
+
+#### _Event Bus / Event Store_
+
+The Event Bus is the mechanism that is used to dispatch events to the subscribed Event Handlers. An Event Store offers the functionality of an Event Bus, with the addition that it persists published events.
+
+The[ Event Bus/Event Store](event-bus-and-event-store.md) section details the capabilities that the Axon Framework provides to help in the development of these components.
+
+_**Event Processors**_
+
+As mentioned above, Event Handlers define the business logic to be performed when an Event is received. Event Processors are the components that take care of the technical aspects of that processing. This includes - Threading Modes, Sequencing Policies, Transaction Management and Error Handling. 
+
+The [Event Processors](event-processors.md) section details the capabilities that the Axon Framework provides to help in the development and configuration of these components.
+
+In both cases, these components are backed by an actual physical infrastructure \(e.g. [Axon Server ](../../axon-server.md)or Other Servers\)
+
+A summary of the representation of the non-functional components is depicted below.
+
+![](../../.gitbook/assets/axon-event-infrastructure.jpg)
+
+## Event Versioning
+
+In the lifecycle of an Axon application, events _will_ typically change their format. As events are stored indefinitely the application should be able to cope with several versions of an event. The [Event Versioning ](versioning-events.md)section will discuss what to keep in mind when developing your events, for backwards \(and forward\) compatibility. It will deep-dive into the upcasting process provided by the Axon Framework to help achieve this concern
```

Comment:
```suggestion
In the lifecycle of an Axon application, events _will_ typically change their format. As events are stored indefinitely the application should be able to cope with several versions of an event. The [Event Versioning](versioning-events.md) section will discuss what to keep in mind when developing your events, for backwards \(and forward\) compatibility. It will deep-dive into the upcasting process provided by the Axon Framework to help achieve this concern.
```